valhalla/sif/costfactory.h

```diff
@@ -53,41 +68,61 @@ template <class cost_t> class CostFactory {
       throw std::runtime_error("No costing provided to cost factory");
 
     // create the cost using the creation function
-    return Create(options.costing(), options);
+    auto costing_index = static_cast<int>(options.costing());
+    if (costing_index < options.costing_options_size()) {
+      return Create(options.costing_options(costing_index));
```

Comment:
now that we have the costing name/type in the costing options object we can just send the costingoptions